addons/mail/static/src/js/chatter.js

```diff
@@ -90,7 +90,7 @@ var Chatter = Widget.extend({
         this._$topbar = this.$('.o_chatter_topbar');
         if(!this._disableAttachmentBox) {
             this.$('.o_topbar_right_area').append(QWeb.render('mail.chatter.Attachment.Button', {
-                count: this.record.data.message_attachment_count || 0,
+                count: this.record.data.message_attachment_count || '',
```

Comment:
I prefer that the button has an option to display the counter, e.g. with argument `displayCounter` as `{boolean}`